players/computer.py

```diff
@@ -1,9 +1,50 @@
-import random
+import copy
 from players.player import Player
+from game_rules import GameRules
 
 class Computer(Player):
     def __init__(self, marker):
         Player.__init__(self, marker)
+        self.rules = GameRules()
 
     def choose_move(self, board):
-        return random.choice(board.available_spots())
+        opponent = self.next_player(self.marker)
+        self.best_move(board, opponent)
+        return self.best_move_var
+
+    def move_score(self, board, last_move, depth):
+        if self.rules.win(board, last_move) and last_move == self.marker:
+            return 10 - depth
+        if self.rules.win(board, last_move) and last_move != self.marker:
+            return depth - 10
+        if self.rules.tie(board):
+            return 0
+
+    def next_player(self, last_move):
+        if last_move == 'X':
+            return 'O'
+        else:
+            return 'X'
+
+    def best_move(self, board, last_move, depth = 0):
+        moves = []
+        scores = []
+        current_move = None
+        if self.rules.game_over(board, last_move):
+            return self.move_score(board, last_move, depth)
+        
+        for spot in board.available_spots():
+            potential_board = copy.deepcopy(board)
+            current_move = self.next_player(last_move)
+            potential_board.insert_value(spot, current_move)
+            scores.append(self.best_move(potential_board, current_move, depth + 1))
+            moves.append(spot)
+            potential_board.insert_value(spot, spot)
```

Comment:
I think the second argument is supposed to be a marker?

Reply:
no, because it is a step when the board duplicate is reset to its previous state after an evaluation of one possible move has been made

Reply:
naming clarification added in https://github.com/olgaboiar/ttt_python/pull/7/commits/214b1c92843637bea8bed54f785f527bf8bc5932